sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -377,8 +377,8 @@ _az_http_client_curl_setup_url(CURL * p_curl, az_http_request const * p_request)
   az_span writable_buffer;
   {
     // set URL as 0-terminated str
-    size_t const url_final_size
-        = az_span_length(p_request->_internal.url) + az_span_length(AZ_SPAN_FROM_STR("\0"));
+    size_t const url_final_size = (size_t)az_span_length(p_request->_internal.url)
```

Comment:
I would change this size_t to a int32_t and then remove all the casting.
I'd also just add 1 instead of creating a span of "\0" and then getting its length; just add a comment about what your doing and shrink the code by several instructions.

Reply:
Done.